File: dgdg-api/src/main/java/com/depromeet/dgdg/config/auth/UserController.java
package com.depromeet.dgdg.config.auth;

import com.depromeet.dgdg.common.ErrorCode;
import com.depromeet.dgdg.common.exception.NotFoundException;
import com.depromeet.dgdg.domain.domain.User.User;
import com.depromeet.dgdg.domain.domain.User.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;


@RestController
@RequiredArgsConstructor
public class UserController {

    private UserRepository userRepository;

    @GetMapping("/auth/authorization/kakao")
    public User findUserById(String socialId) {
        User user = userRepository.findUserById(socialId);
        if (user == null) {
            throw new NotFoundException(String.format("존재하지 않는 아이디입니다."), ErrorCode.NOT_FOUND_EXCEPTION);
        }
        return user;
    }

}

File: dgdg-api/src/main/java/com/depromeet/dgdg/controller/user/UserController.kt
package com.depromeet.dgdg.controller.user

import com.depromeet.dgdg.config.auth.RequiredAuth
import com.depromeet.dgdg.config.auth.UserId
import com.depromeet.dgdg.controller.dto.response.BaseResponse
import com.depromeet.dgdg.service.user.UserService
import com.depromeet.dgdg.service.user.dto.request.UpdateUserInfoRequest
import com.depromeet.dgdg.service.user.dto.response.UserInfoResponse
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.security.SecurityRequirement
import org.springframework.web.bind.annotation.*
import javax.validation.Valid

@RestController
class UserController(
    private val userService: UserService
) {

    @Operation(summary = "유저의 회원 정보를 조회하는 API", security = [SecurityRequirement(name = "BearerKey")])
    @RequiredAuth
    @GetMapping("/api/v1/user/me")
    fun getMyUserInfo(
        @UserId userId: Long
    ): BaseResponse<UserInfoResponse> {
        return BaseResponse.success(userService.getUserInfo(userId))
    }

    @Operation(summary = "유저의 회원 정보를 수정하는 API", security = [SecurityRequirement(name = "BearerKey")])
    @RequiredAuth
    @PutMapping("/api/v1/user/me")
    fun updateMyUserInfo(
        @Valid @RequestBody request: UpdateUserInfoRequest,
        @UserId userId: Long
    ): BaseResponse<UserInfoResponse> {
        return BaseResponse.success(userService.updateUserInfo(request, userId))
    }

    @Operation(summary = "유저의 회원 정보를 삭제하는 API (회원탈퇴)", security = [SecurityRequirement(name = "BearerKey")])
    @DeleteMapping("/api/v1/user/me")
    @RequiredAuth
    fun deleteUser(@UserId userId: Long): BaseResponse<Nothing>{
        userService.deleteUser(userId)
        return BaseResponse.success(null)
    }

}
